lib/mainloop-worker.c

```diff
@@ -103,7 +103,8 @@ _release_thread_id(void)
   g_static_mutex_lock(&main_loop_workers_idmap_lock);
   if (main_loop_worker_id)
     {
-      main_loop_workers_idmap[main_loop_worker_type] &= ~(1 << (main_loop_worker_id - 1));
+      const gint id = main_loop_worker_id & (MAIN_LOOP_MAX_WORKER_THREADS - 1);
```

Comment:
This change will introduces a hidden dependency that MAIN_LOOP_MAX_WORKER_THREADS is the power of 2. I suggest to use the type of the main_loop_worker_idmap (guint64) explicitly hear. i.e.: sizeof(guint64)*8 -1

Note: This bitmask still allows (in either case) to have bigger number than MAIN_LOOP_MAX_WORKER_THREADS,  but we limit its value in: https://github.com/balabit/syslog-ng/blob/19ee6f82d11c0bd98f6cb9a26c6b60dee7a40d03/lib/mainloop-io-worker.c#L87